app/src/marketplace/containers/EditProductPage/SharedSecretEditor.jsx

```diff
@@ -0,0 +1,83 @@
+// @flow
+
+import React, { useEffect, useCallback, useState } from 'react'
+import cx from 'classnames'
+import { I18n } from 'react-redux-i18n'
+
+import KeyField from '$userpages/components/KeyField'
+import AddKeyField from '$userpages/components/KeyField/AddKeyField'
+import useProduct from '../ProductController/useProduct'
+import { getSecrets, postSecret, putSecret, deleteSecret } from '$mp/modules/communityProduct/services'
+import type { Secret } from '$mp/modules/communityProduct/types'
+
+import styles from './sharedSecretEditor.pcss'
+
+type Props = {
+    className?: string
+}
+
+const SharedSecretEditor = ({ className }: Props) => {
+    const product = useProduct()
+    const [secrets, setSecrets] = useState([])
+    const communityId = (product && product.beneficiaryAddress) || ''
+
+    const fetchSecrets = useCallback(async () => {
+        try {
+            const result = await getSecrets(communityId)
+            setSecrets(result)
+        } catch (e) {
+            console.error('Could not load shared secrets', e)
+        }
+    }, [communityId])
+
+    useEffect(() => {
+        if (communityId) {
+            fetchSecrets()
+        }
+    }, [communityId, fetchSecrets])
+
+    return (
+        <div id="shared-secrets" className={cx(className)}>
```

Comment:
Actually now that I think about, the `id` should be removed here and put to `SharedSecrects` component